src/cmdline_manager.ts

```diff
@@ -136,32 +129,24 @@ export class CommandLineManager implements Disposable {
     }
 
     private onAccept = async (): Promise<void> => {
-        await this.onChangeDebouncedInner.flush();
         if (!this.ignoreAcceptEvent) {
             await this.main.client.input("<CR>");
         }
         this.ignoreAcceptEvent = false;
     };
 
     private onChange = async (text: string): Promise<void> => {
-        this.redrawExpected = false;
-        const toType = calculateInputAfterTextChange(this.lastTypedText, text);
-        if (toType !== "") {
-            logger.debug(`onChange: sending cmdline to nvim: "${this.lastTypedText}" + "${toType}" -> "${text}"`);
-            await this.main.client.input(toType);
-            this.lastTypedText = text;
-        } else {
-            logger.debug(`onChange: skip sending cmdline to nvim: "${this.lastTypedText}"`);
-        }
-    };
-
-    private onChangeDebouncedInner = debounce(this.onChange, 100, { leading: false, trailing: true });
-
-    private onChangeDebounced = (text: string): void => {
-        this.onChangeDebouncedInner(text);
-        if (!this.redrawExpected) {
-            this.onChangeDebouncedInner.flush();
+        if (this.pendingNvimUpdates) {
+            this.pendingNvimUpdates--;
```

Comment:
I don't think this could happen, but we could stand to be a bit defensive here and do `this.pendingNvimUpdates = Math.max(0, this.pendingNvimUpdates)`. If `this.pendingNvimUpdates` were to somehow get negative, we would continue taking this branch forever (since all non-zero values are truthy)